### PR TITLE
CIV-6514 CCJ to Formalise payment Screen - Fixed Costs

### PR DESCRIPTION
**Pull request by kenneth-hmcts**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-6514

### Change description ###
This PR contains the changes for the fixed cost page in the LR ITP journey

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java [hunk 1]
```diff
@@ -530,39 +532,63 @@ private boolean checkPastDateValidation(LocalDate localDate) {
 
     private CallbackResponse validateAmountPaid(CallbackParams callbackParams) {
         CaseData caseData = callbackParams.getCaseData();
-        CaseData.CaseDataBuilder<?, ?> updatedCaseData = caseData.toBuilder();
         List<String> errors = new ArrayList<>();
-        if (caseData.getCcjPaymentPaidSomeAmount() != null && caseData.getCcjPaymentPaidSomeAmount()
+        if (caseData.getCcjPaymentDetails().getCcjPaymentPaidSomeAmount() != null
+            && caseData.getCcjPaymentDetails().getCcjPaymentPaidSomeAmount()
```
Comment (TatsyMaslova) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java:537]: perhaps make it more readable, i.e. extract it into a boolean method that would describe what it is checking?

File: src/main/java/uk/gov/hmcts/reform/civil/model/CaseDataParent.java [hunk 2]
```diff
@@ -335,21 +336,8 @@ public class CaseDataParent implements MappableObject {
...
-    @JsonFormat(shape = JsonFormat.Shape.STRING)
-    private final BigDecimal ccjJudgmentAmountClaimFee;
-    @JsonFormat(shape = JsonFormat.Shape.STRING)
-    private final BigDecimal ccjPaymentPaidSomeAmountInPounds;
-    @JsonFormat(shape = JsonFormat.Shape.STRING)
-    private final BigDecimal ccjJudgmentSummarySubtotalAmount;
-    @JsonFormat(shape = JsonFormat.Shape.STRING)
-    private final BigDecimal ccjJudgmentTotalStillOwed;
-    @JsonFormat(shape = JsonFormat.Shape.STRING)
-    private final BigDecimal ccjJudgmentAmountInterestToDate;
```
Comment (TatsyMaslova) [inline src/main/java/uk/gov/hmcts/reform/civil/model/CaseDataParent.java:352]: This is a really good refactor. but are those fields used in any other projects? will we affect them by this refactor?

Comment (kenneth-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/model/CaseDataParent.java:352]: The ccj part is introduced in another ticket CIV-6000, which is currently is LR ITP changes and they are newly added field. So it should not affect other project

**Review by neeta-hmcts — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java (hunk 1 above)
Comment (kenneth-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java]: Update with a new method

**Review by PaulH-HMCTS — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java [hunk 3]
```diff
@@ -530,39 +532,67 @@ private boolean checkPastDateValidation(LocalDate localDate) {
...
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(updatedCaseData.build().toMap(objectMapper))
+            .build();
+    }
+
+    private BigDecimal setUpFixedCostAmount(BigDecimal claimAmount, CaseData caseData) {
+        if (!YES.equals(caseData.getCcjPaymentDetails().getCcjJudgmentFixedCostOption())) {
+            return ZERO;
+        }
+        if (claimAmount.compareTo(BigDecimal.valueOf(25)) < 0) {
```
[the remark this answers is not in the document]
Comment (kenneth-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java]: Updated by using new created constants

**Review by TatsyMaslova — DISMISSED** [review]

**Review by PaulH-HMCTS — APPROVED** [review]

**Review by neeta-hmcts — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java [hunk 4]
```diff
@@ -563,40 +570,67 @@ private boolean checkPastDateValidation(LocalDate localDate) {
...
+
+        updatedCaseData.ccjPaymentDetails(ccjPaymentDetails);
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(updatedCaseData.build().toMap(objectMapper))
+            .build();
+    }
+
+    private BigDecimal setUpFixedCostAmount(BigDecimal claimAmount, CaseData caseData) {
+        if (!YES.equals(caseData.getCcjPaymentDetails().getCcjJudgmentFixedCostOption())) {
```
Comment (TatsyMaslova) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java:619]: I am not sure whether this should be in the handler. the handler is big already. and  has too many responsibilities. can we move it somewhere outside the handler and then the handler can use it?

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java [hunk 5]
```diff
@@ -563,40 +570,67 @@ private boolean checkPastDateValidation(LocalDate localDate) {
...
+            return lowCostAmount;
+        } else if (claimAmount.compareTo(upperRangeClaimAmount) <= 0) {
+            return midCostAmount;
+        } else {
+            return highCostAmount;
+        }
+    }
+
+    private boolean isPaidSomeAmountMoreThanClaimAmount(CaseData caseData) {
+        return caseData.getCcjPaymentDetails().getCcjPaymentPaidSomeAmount() != null
```
Comment (TatsyMaslova) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java:632]: same applies to this method. could be inside casedata instead?

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java (hunk 4 above)
Comment (kenneth-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java]: Updated

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java (hunk 5 above)
Comment (kenneth-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java]: Updated

**Review by TatsyMaslova — APPROVED** [review]

**Review by PaulH-HMCTS — APPROVED** [review]

**Pull request merged**